packages/react/src/components/Button/index.tsx

```diff
@@ -1,42 +1,31 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { ButtonHTMLAttributes, forwardRef, Ref } from 'react';
 import classNames from 'classnames';
 
-export interface ButtonProps
-  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  buttonRef?: Ref<HTMLButtonElement>;
   variant?: 'primary' | 'secondary' | 'error' | 'link';
-  buttonRef?: React.Ref<HTMLButtonElement>;
-  disabled?: boolean; // todo, investigate why disabled is needed
   thin?: boolean;
 }
 
-export default class Button extends React.Component<ButtonProps> {
-  static defaultProps = {
-    variant: 'primary',
-    buttonRef: () => {}
-  };
-
-  static propTypes = {
-    variant: PropTypes.oneOf(['primary', 'secondary', 'error', 'link']),
-    children: PropTypes.node,
-    className: PropTypes.string,
-    buttonRef: PropTypes.oneOfType([
-      PropTypes.func,
-      PropTypes.shape({ current: PropTypes.any })
-    ])
-  };
-
-  static displayName = 'Button';
-
-  render() {
-    const {
-      variant,
+const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      variant = 'primary',
+      thin,
       children,
       className,
       buttonRef,
-      thin,
       ...other
-    } = this.props;
+    }: ButtonProps,
+    ref
+  ) => {
+    if (buttonRef) {
+      console.warn(
+        "%c Warning: 'buttonRef' prop is deprecated, please use 'ref'. ",
+        'background: #222; color: #bada44'
+      );
+    }
```

Comment:
Let's leave out the `console.warn` here.

Reply:
removed in the latest commit